exasol_sphinx_github_pages_generator/generate_index.py

```diff
@@ -0,0 +1,268 @@
+    be used in context "quotes" ="another_dict/_static/doctools.js"'
+
+    will be turned into
+
+    'this text contains "source_branch/_static" multiple times because "source_branch/_static" is a keyword'
+    and can be used in context "quotes" ="source_branch/another_dict/_static/doctools.js"'
+
+    :param original_line: String that should be changed to include "source_branch".
+    :param source_branch: string that should be inserted before "_static".
+    :return: altered input string with the inclusion of "source_branch".
+    """
+    static_position_list = [m.start() for m in re.finditer("_static", original_line)]
+    double_quote_list = [m.start() for m in re.finditer('"', original_line)]
+    new_line = original_line
+    # find where new path needs to be inserted
+    quote_pos = [find_quote_pos(static_pos, double_quote_list) for static_pos in static_position_list]
+    # insert in reverse order, so the positions don't need to be adjusted after each insertion
+    for pos in quote_pos[::-1]:
+        new_line = new_line[:pos+1] + f'{source_branch}/' + new_line[pos+1:]
+    return new_line
+
+
+def get_meta_lines(index_path: Path, source_branch: str) -> List[str]:
+    """
+    Given the path to a file, reads each line of the file, and extracts meta_lines. Aborts if source_branch is an
+    empty string. If a meta_line contains the
+    keyword "_static", the given source_branch string is added to the line using the "alter_meta_line" function.
+
+    This is done because Sphinx puts all style sheets or script files for themes in the "_static"
+    directory inside the build-output directory. We want to use these to make the release-index the
+    same/similar style as the rest of the documentation.
+    So we steal the lines describing them from the source-branch/index.html,
+    but since the new index.html is in another directory, we need to adjust the paths.
+
+    Meta_lines are defined as lines containing the keywords "_static" or "<meta".
+
+    :param index_path: Path to a file to be read.
+    :param source_branch: String to be inserted into the meta_lines.
+    :return: List of strings containing all extracted and altered meta_lines.
+    """
+    print("get_meta_lines")
+    if not source_branch:
+        raise ValueError("No source branch was given to get_meta_lines")
+    with open(index_path) as file:
+        meta_lines = []
+        for line in file.readlines():
+            if "_static" in line or "<meta" in line:
+                line = alter_meta_line(line, source_branch)
+                meta_lines.append(line)
+    return meta_lines
+
+
+def get_footer(index_path: Path) -> List[str]:
+    """
+    Given the path to a file, reads each line of the file, and extracts the footer which Sphinx generates if it exists.
+    We want to use this footer to make the release-index page of a similar style as the rest of the documentation.
+    So we steal the lines describing tefooter from source-branch/index.html.
```

Comment:
wooh, this and get_meta_lines is a bit overengineered, and I fear it will be quite fragile.